.github/workflows/jira-creation.yml

```diff
@@ -0,0 +1,51 @@
+# **what?**
+# Mirrors issues into Jira. Includes the information: title
+# description, GitHub Issue ID and URL
+
+# **why?**
+# Jira is our tool for tracking and we need to see these issues in there
+
+# **when?**
+# On issue creation or when an issue is labeled `Jira`
+
+name: Jira Issue Creation
+
+on:
+  issues:
+    types: [opened, labeled]
+    
+permissions:
+  issues: write
+
+jobs:
+  create-issue:
+    runs-on: ubuntu-latest
+    if: |
+      ((github.event.action == 'opened') || 
+      (github.event.action == 'labeled' && github.event.label.name == 'jira'))
+    steps: 
+      - name: Login
+        uses: atlassian/gajira-login@v2.0.0
+        env:
+          JIRA_BASE_URL: ${{ secrets.JIRA_BASE_URL }}
+          JIRA_USER_EMAIL: ${{ secrets.JIRA_USER_EMAIL }}
+          JIRA_API_TOKEN: ${{ secrets.JIRA_API_TOKEN }}
+          
+      - name: Create JIRA Ticket
+        id: create
+        uses: atlassian/gajira-create@v2.0.1
+        with:
+          project: CORE
+          issuetype: "GitHub Issue"
+          summary: '${{ github.event.issue.title }}'
+          description: '${{ github.event.issue.body }}'
+          fields: '{ "labels": ["core"], "customfield_10045": "${{ github.event.issue.html_url }}", "customfield_10046": ${{ github.event.issue.number }} }'
+ 
+      - name: Update Issue Title
+        run: |
+          curl --request PATCH \
+          --url https://api.github.com/repos/${{ github.repository }}/issues/${{ github.event.issue.number }} \
+          --header 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' \
+          --header 'content-type: application/json' \
+          --header 'Accept: application/vnd.github.v3+json' \
+          --data '{ "title": "[${{ steps.create.outputs.issue }}] ${{ github.event.issue.title }}" }'
```

Comment:
if we split out the above title parsing logic into its own action it would be nice for the creation logic to live along side it so that we don't have to hunt down all the usages of serialization and deserialization if we ever want to change something.

Reply:
My current thought is to work on the other workflows (i.e. label mirroring, assignee mirroring) next and then identify what I'm copying and pasting and then look at the refactor out into their own actions then. It makes sense to do that but it isn't the easiest thing to create these customer actions so I want to make sure we really do need them before I start factoring them out

Reply:
Sounds reasonable to me.